src/CodeGeneration/OfferFriendlyOverloadsGenerator.cs

```diff
@@ -110,15 +110,23 @@ select new
                     .WithSemicolonToken(SyntaxFactory.Token(SyntaxKind.None))
                     .WithExpressionBody(null);
 
-                var flags = GeneratorFlags.NativePointerToIntPtr;
-                var intPtrOverload = transformedMethodBase
-                    .WithParameterList(TransformParameterList(method.ParameterList, flags, parametersToFriendlyTransform))
-                    .WithBody(CallNativePointerOverload(method, flags, parametersToFriendlyTransform));
-                generatedType = generatedType.AddMembers(intPtrOverload);
+                var preexistingMethod = from newMethod in generatedType.Members.OfType<MethodDeclarationSyntax>()
+                                        where WhereIsIntPtrParameter(method.ParameterList.Parameters, newMethod.ParameterList.Parameters).Any() && newMethod.Identifier.ValueText == transformedMethodBase.Identifier.ValueText
+                                        select method;
+
+                if (!preexistingMethod.Any())
+                {
+                    var flags = GeneratorFlags.NativePointerToIntPtr;
+                    var intPtrOverload = transformedMethodBase
+                        .WithParameterList(TransformParameterList(method.ParameterList, flags, parametersToFriendlyTransform))
+                        .WithBody(CallNativePointerOverload(method, flags, parametersToFriendlyTransform));
+                    generatedType = generatedType.AddMembers(intPtrOverload);
```

Comment:
From @vbfox's [comment](https://github.com/AArnott/pinvoke/pull/291#discussion_r79999355):

> Can't this be made more generic ?
Like a `AddMembersIfNotExists` that would work in all cases, avoiding duplication of methods with the same name and the same parameter types.

> I suggest we create a method doing AddMember only if no conflicting overload exists.

> So instead of computing "should we generate that overload" and adding the method or not we always use that AddMember variant and let it handle doing it or not.

From @AArnott's [comment](https://github.com/AArnott/pinvoke/pull/291#discussion_r80363669):

> I like the idea of wrapping up the `AddIfNotExists` behavior as a method as well so we can focus on that and getting it right.